storage/remote/generic/generic.go

```diff
@@ -62,16 +69,15 @@ func (c *Client) Store(samples model.Samples) error {
 		}
 		req.Timeseries = append(req.Timeseries, ts)
 	}
-	client := NewGenericWriteClient(c.conn)
 	ctxt, _ := context.WithTimeout(context.Background(), c.timeout)
```

Comment:
It's just how it's supposed to be used. Might not be necessary here (no idea, tbh) but its how context receivers expect things to be done if the caller behaves according to the documentation: https://godoc.org/context#WithCancel

```
Canceling this context releases resources associated with it, so code should call cancel as soon as the operations running in this Context complete.
```

I think it's very easy to forget an the call might have spawned a bunch of goroutines, which might be dangling without the context being canceled. Better to make this a best practice everywhere.